core/src/main/java/org/apache/iceberg/BaseFile.java

```diff
@@ -277,6 +280,9 @@ public void put(int i, Object value) {
         this.sortOrderId = (Integer) value;
         return;
       case 17:
+        this.schemaId = (int) value;
```

Comment:
I updated the order because we add the `ROW_POSITION` after the schema fields of DataFile here: https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/BaseFile.java#L99. Some experts could give some advice for this update.